lib/fleiss/backend/active_record/concern.rb

```diff
@@ -79,13 +79,23 @@ def start(owner, now: Time.zone.now)
         # @param [String] owner
         # @return [Boolean] true if successful.
         def finish(owner, now: Time.zone.now)
-          with_lock do
+          with_isolation do
             self.class
                 .in_progress(owner)
                 .where(id: id)
                 .update_all(finished_at: now) == 1
           end
         end
+
+        private
+
+        def with_isolation(&block)
+          args = []
+          args.push(isolation: :repeatable_read) if self.class.connection.supports_transaction_isolation?
+          transaction(*args, &block)
+        rescue ::ActiveRecord::SerializationFailure
+          false
```

Comment:
Are you sure it should be suppressed? I'd better re-raise it not to surprise lib consumers.